pkg/sessionctx/variable/session.go

```diff
@@ -1926,6 +1927,48 @@ func (p *PlanCacheParamList) AllParamValues() []types.Datum {
 	return p.paramValues
 }
 
+// LazyStmtText represents the sql text of a stmt that used in log. It's lazily evaluated to reduce the mem allocs.
+type LazyStmtText struct {
+	text   *string
+	SQL    string
+	Redact string
+	Params PlanCacheParamList
+	Format func(string) string
+}
+
+// SetText sets the text directly.
+func (s *LazyStmtText) SetText(text string) {
```

Comment:
How about 
```
func (s *LazyStmtText) SetText(text *string) {
   s.text = text
}
```

Reply:
Not much difference I think.
![2024-06-21_174628](https://github.com/pingcap/tidb/assets/6850317/6234f8cd-661c-4d80-895e-fcaeb2b6986b)